core/shared/src/main/scala/org/virtuslab/yaml/NodeVisitor.scala

```diff
@@ -52,7 +68,32 @@ class NodeVisitor(node: Node, selectors: List[NodeSelector]):
       case _ => Left(TraverseError(s"Expeceted index of sequence, insted found end of path"))
     }
 
-  private def updateMappingNode(value: String, mapping: MappingNode) =
+  private def removeSequenceNode(
+      sequence: SequenceNode
+  ): Either[TraverseError, SequenceNode] =
+    selectors match {
+      case IntSelector(index) :: rest =>
+        if (rest.isEmpty) {
+          Right(
+            sequence.copy(
+              nodes = sequence.nodes.patch(index, Nil, 1)
+            )
+          )
+        } else {
+          val nodes      = sequence.nodes
+          val updateNode = NodeVisitor(nodes(index), rest).removeValue()
+          updateNode.map(node => sequence.copy(nodes = nodes.updated(index, node)))
+        }
+      case StringSelector(field) :: rest =>
+        Left(
+          TraverseError(
+            s"Expeceted index of sequence, insted found string path: ${field}"
```

Comment:
`Expeceted` -> `Expected`, `insted` -> `instead`
Also, maybe following? 
```suggestion
            s"Found string path: ${field} traversing sequence, but index was expected"
```